sql/core/src/main/scala/org/apache/spark/sql/execution/systemcatalog/InformationSchema.scala

```diff
@@ -0,0 +1,312 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.systemcatalog
+
+import scala.collection.mutable.ArrayBuffer
+
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql._
+import org.apache.spark.sql.catalyst.catalog.SessionCatalog.{DEFAULT_DATABASE, INFORMATION_SCHEMA_DATABASE}
+import org.apache.spark.sql.catalyst.dsl.plans._
+import org.apache.spark.sql.catalyst.expressions._
+import org.apache.spark.sql.catalyst.plans.logical.Project
+import org.apache.spark.sql.execution.datasources._
+import org.apache.spark.sql.internal.CatalogImpl
+import org.apache.spark.sql.sources._
+import org.apache.spark.sql.types._
+
+/**
+ * INFORMATION_SCHEMA is a database consisting views which provide information about all of the
+ * tables, views, columns in a database.
+ *
+ * These views are designed to be populated by this package in order to be independent from
+ * Spark catalog. To keep minimal dependency, currently INFORMATION_SCHEMA views are implemented as
+ * Spark temporary views with a database prefix: `SessionCatalog.INFORMATION_SCHEMA_DATABASE`.
+ *
+ * The following is the class hierarchy in this package rooted at InformationSchemaRelationProvider.
+ *
+ * InformationSchemaRelationProvider
+ *  -> DatabasesRelationProvider
+ *  -> TablesRelationProvider
+ *  -> ViewsRelationProvider
+ *  -> ColumnsRelationProvider
+ *  -> SessionVariablesRelationProvider
+ */
+
+/**
+ * InformationSchema object provides bootstrap and utility functions.
+ */
+object InformationSchema {
+
+  /**
+   * Register INFORMATION_SCHEMA database. SessionCatalog.catalog invokes this function.
+   */
+  def registerInformationSchema(sparkSession: SparkSession): Unit = {
+    sparkSession.sql(s"CREATE DATABASE IF NOT EXISTS $INFORMATION_SCHEMA_DATABASE")
```

Comment:
I am lost. A database `INFORMATION_SCHEMA_DATABASE` is created through a SQL command, but the function `databaseExists` always returns true for the database for `INFORMATION_SCHEMA_DATABASE`.